core/src/test/java/org/elasticsearch/search/aggregations/bucket/DoubleTermsIT.java

```diff
@@ -359,6 +362,32 @@ private void testIncludeExcludeResults(double[] includes, double[] excludes, dou
             assertThat(bucket.getDocCount(), equalTo(1L));
         }
     }
+    
+    public void testSingleValueFieldWithPartitionedFiltering() throws Exception {
+        runTestFieldWithPartitionedFiltering(SINGLE_VALUED_FIELD_NAME);
+    }
+    
+    public void testMultiValueFieldWithPartitionedFiltering() throws Exception {
+        runTestFieldWithPartitionedFiltering(MULTI_VALUED_FIELD_NAME);
+    }
+    
+    private void runTestFieldWithPartitionedFiltering(String field) throws Exception {
+        final int numPartitions = randomIntBetween(1, 4);
+        Set<Number> foundTerms = new HashSet<>();
+        for (int partition = 1; partition <= numPartitions; partition++) {
+            SearchResponse response = client().prepareSearch("idx").setTypes("type").addAggregation(terms("terms").field(field)
+                    .includeExclude(new IncludeExclude(partition, numPartitions)).collectMode(randomFrom(SubAggCollectionMode.values())))
+                    .execute().actionGet();
+            assertSearchResponse(response);
+            Terms terms = response.getAggregations().get("terms");
+            assertThat(terms, notNullValue());
+            assertThat(terms.getName(), equalTo("terms"));
+            for (Bucket bucket : terms.getBuckets()) {
+                assertFalse(foundTerms.contains(bucket.getKeyAsNumber()));
+                foundTerms.add(bucket.getKeyAsNumber());
+            }
+        }
```

Comment:
maybe we should also assert on the size of `foundTerms` to ensure we visited all of them?